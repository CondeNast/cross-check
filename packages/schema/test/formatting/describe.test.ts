import { subject, setupSchemaTest, teardownSchemaTest } from "../support";

describe("[schema] formatting - describe", () => {
  beforeEach(setupSchemaTest);
  afterEach(teardownSchemaTest);

  test("simple", () => {
    let { format } = subject();
    expect(
      format.published.describe("SimpleArticle")
    ).toMatchSnapshot();
  });

  test("detailed", () => {
    let { format } = subject();
    expect(
      format.published.describe("MediumArticle")
    ).toMatchSnapshot();
  });

  test("relationships", () => {
    let { format } = subject();
    expect(
      format.published.describe("Related")
    ).toMatchSnapshot();
  });

  test("nested", () => {
    let { format } = subject();
    expect(
      format.published.describe("Nesting")
    ).toMatchSnapshot();
  });
});
