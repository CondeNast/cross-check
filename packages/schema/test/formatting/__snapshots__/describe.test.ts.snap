// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[schema] formatting - describe detailed 1`] = `
"{
  hed: <single line string>,
  dek?: <string>,
  body: <string>,
  author?: {
    first?: <single line string>,
    last?: <single line string>
  },
  issueDate?: <ISO Date>,
  canonicalUrl?: <url>,
  tags?: list of <single word string>,
  categories: list of <single line string>,
  geo?: {
    lat: <integer>,
    long: <integer>
  },
  contributors?: list of {
    first?: <single line string>,
    last?: <single line string>
  },
  relatedArticles?: has many MediumArticle
}"
`;

exports[`[schema] formatting - describe nested 1`] = `
"{
  people: list of {
    first?: <single line string>,
    last?: <string>
  }
}"
`;

exports[`[schema] formatting - describe relationships 1`] = `
"{
  first?: <single line string>,
  last?: <string>,
  person: has one SimpleArticle,
  articles?: has many MediumArticle
}"
`;

exports[`[schema] formatting - describe simple 1`] = `
"{
  hed: <single line string>,
  dek?: <string>,
  body: <string>
}"
`;
